### PR TITLE
python3Packages.xdg: init at 4.0.1

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
one of the scripts at work required this. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
[1 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/83389
21 package marked as broken and skipped:
octave-jit octoprint php72Packages-unit.php_excel php72Packages.php_excel php73Packages-unit.php_excel php73Packages.php_excel php74Packages-unit.couchbase php74Packages-unit.pcs php74Packages-unit.php_excel php74Packages.couchbase php74Packages.php_excel python27Packages.habanero python27Packages.handout python27Packages.hass-nabucasa python37Packages.notify python37Packages.pasteScript python37Packages.pyblock python38Packages.notify python38Packages.pasteScript python38Packages.pyblock qes

16 package built:
python27Packages.cleo python27Packages.clikit python27Packages.pastel python27Packages.poetry python37Packages.aria2p python37Packages.cleo python37Packages.clikit python37Packages.pastel python37Packages.poetry python37Packages.xdg python38Packages.aria2p python38Packages.cleo python38Packages.clikit python38Packages.pastel python38Packages.poetry python38Packages.xdg
```

### CODE REVIEW

File: pkgs/development/python-modules/poetry/default.nix [hunk 1]
```diff
@@ -36,7 +36,7 @@ in buildPythonPackage rec {
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0y528095njf28isbcp5iwbf12j67xhxnrkac93pws0zy133v7kc1";
+    sha256 = "02h387k0xssvv78yy82pcpknpq4w5ym2in1zl8cg9r5wljl5w6cf";
```
Comment (teto) [inline pkgs/development/python-modules/poetry/default.nix]: why this change ?

Comment (jonringer) [reply pkgs/development/python-modules/poetry/default.nix]: because it's wrong for me. Try doing `nix-build -A python3Packages.poetry.src --check` and you should get the same

Comment (jonringer) [reply pkgs/development/python-modules/poetry/default.nix]: hydra also agrees https://hydra.nixos.org/eval/1577793?filter=poetry&compare=1577743&full=

Comment (jonringer) [reply pkgs/development/python-modules/poetry/default.nix]: https://hydra.nixos.org/build/115163885/nixlog/1

Comment (jonringer) [reply pkgs/development/python-modules/poetry/default.nix]: likely, poetry re-uploaded the poetry package with different contents. I don't think pypi is immutable (at least I've seen removals before)

Comment (teto) [reply pkgs/development/python-modules/poetry/default.nix]: I checked indeed. Sad it's not immutable :( I hope it's a LGTM update on their side, it doesn't seem possible to check the different uploads from https://pypi.org/project/poetry/1.0.5/

**Comment by teto** [comment]
@GrahamcOfBorg build python3Packages.xdg python3Packages.pastel python3Packages.clikit
